nixos/modules/programs/_1password.nix

```
@@ -0,0 +1,46 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.programs._1password;
+in {
+  options = {
+    programs._1password = {
+      enable = mkEnableOption "The 1Password CLI tool with biometric unlock and integration with the 1Password GUI.";
+
+      groupId = mkOption {
+        type = types.int;
+        example = literalExpression "5001";
+        description = ''
+          The GroupID to assign to the onepassword-cli group, which is needed for integration with the 1Password GUI. The group ID must be 1000 or greater.
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs._1password;
+        defaultText = literalExpression "pkgs._1password";
+        example = literalExpression "pkgs._1password";
+        description = ''
+          The 1Password CLI derivation to use.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
```

I haven't tested this, but something like this should work to ensure that people don't try and use and old version of `1password-cli` that won't work.

```suggestion
  config = mkIf cfg.enable {
    assertions =
      [ 
        { 
          assertion = lib.versionOlder cfg.package.version "2.0.0";
          message = "1password-cli version 2.0.0 or greater is required (version ${cfg.package.version} provided)";
        }
      ];
      
```

---

Is that necessary? I assume there's no plan on backporting it to 21.11 (because backporting 2.0.0 would be a breaking change anyway), and 22.05 is around the corner.

---

¯\_(ツ)_/¯

I don't have a strong opinion here. I also assume there's not plan to backport, so this would only help if someone was overlaying in `1password-cli` from an older version of `nixpkgs`.

---

Since the same breakage is true for `1password-gui` and we didn't safeguard that module, I think this is fine as-is.